script.js

```diff
@@ -0,0 +1,248 @@
+            count.innerText = parseInt(count.innerText) + 1;
+            console.log(count);
+            this.coin.update();
+
+        }
+
+        if (colliding(this.badCoin.location, this.player.location)) {
+            var count = document.getElementById("counter");
+            count.innerText = 0;
+            console.log(count)
+            this.badCoin.update();
+
+        }
+
+        if (colliding(this.badCoinB.location, this.player.location)) {
+            var count = document.getElementById("counter");
+            count.innerText = 0;
+            console.log(count)
+            this.badCoinB.update();
+
+        }
+    }
+}
+
+class Player {
+    constructor(game) {
+        this.game = game;
+        this.screen = game.screen;
+        this.keyboarder = new Keyboarder();
+        this.location = {
+            x: 230,
+            y: 230,
+            sizeX: 80,
+            sizeY: 20
+        }
+    }
+    draw() {
+        this.screen.fillStyle = "#FF8800";
+        var sizeX = this.location.sizeX;
+        var sizeY = this.location.sizeY;
+        var leftX = this.location.x
+        var leftY = this.location.y
+        this.screen.fillRect(leftX, leftY, sizeX, sizeY);
+    }
+    update() {
+
+        if (this.keyboarder.isDown(Keyboarder.KEYS.LEFT) && this.keyboarder.isDown(Keyboarder.KEYS.SPACE) && this.location.x > 155) {
+            this.location.x -= 7;
+
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.RIGHT) && this.keyboarder.isDown(Keyboarder.KEYS.SPACE) && this.location.x < 264) {
+            this.location.x += 7;
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.DOWN) && this.keyboarder.isDown(Keyboarder.KEYS.SPACE) && this.location.y < 325) {
+            this.location.y += 7;
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.UP) && this.keyboarder.isDown(Keyboarder.KEYS.SPACE) && this.location.y > 155) {
+            this.location.y -= 7;
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.LEFT) && this.location.x > 155) {
+            this.location.x -= 3;
+
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.RIGHT) && this.location.x < 264) {
+            this.location.x += 3;
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.DOWN) && this.location.y < 325) {
+            this.location.y += 3;
+        } else if (this.keyboarder.isDown(Keyboarder.KEYS.UP) && this.location.y > 155) {
+            this.location.y -= 3;
+        }
+
+
+    }
+}
+
+class Coin {
+    constructor(game) {
+        this.game = game;
+        this.screen = game.screen;
+        this.location = {
+            x: (Math.floor(Math.random() * 155) + 155),
+            y: ((Math.floor(Math.random() * 155)) + 155)
+        }
+    }
+    draw() {
+        this.screen.fillStyle = "#FFDD00";
+        var sizeX = 20;
+        var sizeY = 20;
+        this.screen.fillRect(this.location.x, this.location.y, sizeX, sizeY);
+    }
+    update() {
```

Comment:
We talked about this in class -- `update` is not updating the coin per tick. Naming is important -- consider naming this something like `reappear` or `move`.